tests/components/livisi/test_config_flow.py

```diff
@@ -66,3 +69,26 @@ async def test_create_entity_after_login_error(
             user_input=VALID_CONFIG,
         )
     assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
+
+
+async def test_reauth_flow(hass: HomeAssistant) -> None:
+    """Test reauthentication flow."""
+    entry: MockConfigEntry = create_entry()
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_REAUTH,
+            "entry_id": entry.entry_id,
+        },
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "reauth_confirm"
+
+    with mocked_livisi_login(), mocked_livisi_controller(), mocked_livisi_setup_entry():
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], VALID_CONFIG
+        )
+        assert result["type"] == data_entry_flow.FlowResultType.ABORT
+        assert result["reason"] == "reauth_successful"
```

Comment:
I think you should also add an intermediate test for a failed reauth:
reauth_confirm step -> (bad-input) -> user step -> (good-input) -> abort/success

Reply:
I added more tests, thank you.